gems/pending/spec/util/mount/miq_generic_mount_session_spec.rb

```diff
@@ -2,25 +2,29 @@
 require 'util/mount/miq_generic_mount_session'
 
 describe MiqGenericMountSession do
-  context "#connect" do
-    before do
-      MiqGenericMountSession.stub(:raw_disconnect)
-      @s1 = MiqGenericMountSession.new({:uri => '/tmp/abc'})
-      @s2 = MiqGenericMountSession.new({:uri => '/tmp/abc'})
+  let(:prefix) { File.join(Dir.tmpdir, "miq_") }
 
-      @s1.logger = Logger.new("/dev/null")
-      @s2.logger = Logger.new("/dev/null")
+  it "#connect returns a string pointing to the mount point" do
+    described_class.stub(:raw_disconnect)
+    s = described_class.new(:uri => '/tmp/abc')
+    s.logger = Logger.new("/dev/null")
+
+    expect(s.connect).to match(%r(\A#{prefix}\d{8}-\d{5}-\w+\z))
+
+    s.disconnect
+  end
+
+  context "#mount_share" do
+    it "without :mount_point uses default temp directory as a base" do
+      expect(described_class.new(:uri => '/tmp/abc').mount_share).to match(%r(\A#{prefix}\d{8}-\d{5}-\w+\z))
     end
 
-    after do
-      @s1.disconnect
-      @s2.disconnect
+    it "with :mount_point uses specified directory as a base" do
+      expect(described_class.new(:uri => '/tmp/abc', :mount_point => "abc").mount_share).to match(%r(\Aabc/miq_\d{8}-\d{5}-\w+\z))
```

Comment:
I don't understand why abc is in the URI and the mount point name.  I think it would be clearer if they were different things as it should make the assertion much easier to make.  I think we only care that we get the `abc` from the mount_point followed by the `/miq_*`